packages/@sanity/form-builder/src/inputs/ObjectInput/ConditionalField.tsx

```diff
@@ -0,0 +1,61 @@
+import React, {useMemo} from 'react'
+import {
+  BaseSchemaType,
+  HiddenOptionCallbackContext,
+  HiddenOptionCallback,
+  SanityDocument,
+} from '@sanity/types'
+
+import withDocument from '../../utils/withDocument'
+
+type HiddenOption = BaseSchemaType['hidden']
+
+function isThenable(value: any) {
+  return typeof value?.then === 'function'
+}
+
+function checkCondition(
+  schemaHiddenOption: HiddenOptionCallback | undefined | boolean,
+  context: HiddenOptionCallbackContext
+): boolean {
+  const result =
+    typeof schemaHiddenOption === 'function' ? schemaHiddenOption(context) : schemaHiddenOption
```

Comment:
it might be a good idea to try catch this result and warn. a common case for this would be users not handling the undefined case on the initial undefined document and in that case i think we can save the editor from crashing without many unwanted effects

Reply:
Great point! Fixed in b2b4797 (I didn't add any state for preventing overlogging for errors thrown in the callback  as the error may differ from each call, but we can consider improving on this later if needed).